nautobot_circuit_maintenance/handle_notifications/sources.py

```diff
@@ -425,6 +425,16 @@ def fetch_email(self, logger: Job, msg_id: bytes, since: Optional[int]) -> Optio
 
         See data format:  https://developers.google.com/gmail/api/reference/rest/v1/users.messages#Message
         """
+
+        def get_raw_payload_from_parts(parts, provider_data_type):
+            """Helper function to extract the raw_payload from a multiple parts via a recursive call."""
+            for part_inner in parts:
+                for header_inner in part_inner["headers"]:
+                    if header_inner.get("name") == "Content-Type" and provider_data_type in header_inner.get("value"):
+                        return self.extract_raw_payload(part_inner["body"], msg_id)
+                    elif header_inner.get("name") == "Content-Type" and "multipart" in header_inner.get("value"):
+                        return get_raw_payload_from_parts(part_inner["parts"], provider_data_type)
```

Comment:
Probably would be good style-wise to have an explicit `return None` at the end of the function if none of the earlier conditions are met.